### PR TITLE
[gitpod-api] guard against unsupported notifications

### LINKED ISSUE #5717: [code] new gitpod-protocol method leads to (non-critical) errors
Recently introduced `onPrebuildUpdate` notification is not understood by the older version of `gitpod-protocol` currently used in code (or Gitpod extension.) Errors like this are logged on frontend whenever a prebuild update event is pushed from server to client. 

```
[Extension Host] Notification handler 'onPrebuildUpdate' failed with message: Cannot read property 'apply' of undefined (at console.e.<computed> [as error] (/ide/extensions/typescript-language-features/dist/extension.js:1:139792))
S @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:35
_doInvokeHandler @ rpcProtocol.ts:472
_invokeHandler @ rpcProtocol.ts:457
_receiveRequest @ rpcProtocol.ts:373
_receiveOneMessage @ rpcProtocol.ts:295
(anonymous) @ rpcProtocol.ts:160
fire @ event.ts:577
fire @ ipc.net.ts:513
_receiveMessage @ ipc.net.ts:871
(anonymous) @ ipc.net.ts:710
fire @ event.ts:577
acceptChunk @ ipc.net.ts:286
(anonymous) @ ipc.net.ts:247
(anonymous) @ browserSocketFactory.ts:197
fire @ event.ts:577
E._fileReader.onload @ browserSocketFactory.ts:81
09:09:28.416 
```

cc. @akosyakov

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

Adding new notifications from Gitpod server to clients is not backward compatible change. We used to have a guard which will allow client ignore such notifications but removed it to support proxies: https://github.com/gitpod-io/gitpod/commit/4c2fab7a93e6d9589d6b14fdfd9cfa1a878e9454#diff-b8847df09095afd1fed8277f763df0e828b40f2e071d73f3bbc4ed8c131903f9 Since `in` operator did not return any properties for proxies.

This PR reintroduces the guard but directly in handling of a concrete notification.

Changes in Gitpod Code: https://github.com/gitpod-io/openvscode-server/commit/9255ed132288a5b3591e4f8ce79fdb83f3fc6f9c

TODO:
- [ ] don't forget to publish new desktop version after PR is approved

## Related Issue(s)

Fixes #5717

## How to test

Enable latest VS Code: https://ak-gp-api-notif-guard.staging.gitpod-dev.com/preferences
Start a workspace to verify that proxying still works: https://ak-gp-api-notif-guard.staging.gitpod-dev.com/#github.com/akosyakov/parcel-demo

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#5717*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jeanp413 — APPROVED** [review]
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 59e73183baef20456342210211cf6c502d83017f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5724#pullrequestreview-756998244" title="LGTM">jeanp413</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#5717*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
/lgtm 

Code changes looks reasonable

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5724#issuecomment-921612499" title="LGTM">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5724#pullrequestreview-756998244" title="LGTM">jeanp413</a>*

Associated issue: *#5717*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [AlexTugarev]
- ~~[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
